iOS_SDK/OneSignalSDK/Source/OneSignalLifecycleObserver.m

```diff
@@ -0,0 +1,115 @@
+/**
+Modified MIT License
+
+Copyright 2020 OneSignal
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+1. The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+2. All copies of substantial portions of the Software may only be used in connection
+with services provided by OneSignal.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#import <Foundation/Foundation.h>
+#import "OneSignalLifecycleObserver.h"
+#import "OneSignal.h"
+#import "OneSignalCommonDefines.h"
+#import "OneSignalTracker.h"
+#import "OneSignalLocation.h"
+#import "OSMessagingController.h"
+
+@implementation OneSignalLifecycleObserver
+
+static OneSignalLifecycleObserver* _instance = nil;
+
++(OneSignalLifecycleObserver*) sharedInstance {
+    @synchronized( _instance ) {
+        if( !_instance ) {
+            _instance = [[OneSignalLifecycleObserver alloc] init];
+        }
+    }
+    
+    return _instance;
+}
+
++ (BOOL)isAppUsingUIScene {
+    if (@available(iOS 13.0, *)) {
+        return [[NSBundle mainBundle] objectForInfoDictionaryKey:@"UIApplicationSceneManifest"] != nil;
```

Comment:
Confirmed with Apple's docs this is the source of truth if the app is setup as a scene.

> The presence of this key indicates that the app supports scenes and does not use an app delegate object to manage transitions to and from the foreground or background.
https://developer.apple.com/documentation/bundleresources/information_property_list/uiapplicationscenemanifest